Build occasionally fails when using shared GitInfo.cache

## Describe the Bug

When the `$(GitCachePath)` is set to "use a shared location for different projects", as described in the documentation, occasionally the build will fail, at least on the build server.

## Steps to Reproduce

Create a solution with multiple projects in it.

Define a shared location for the GitInfo.cache, something like this f.e.
```xml
<GitCachePath>$(SolutionDir)/obj/</GitCachePath>
```

Build using `msbuild /m`, while having some changes in the repository, so GitInfo.cache would be updated.

## Expected Behavior

<!-- Describe what you expected to happen. -->
Works every time.

## Actual Behavior

<!-- If you see an exception, put the WHOLE THING here. -->

Occasionally, it doesn't work.

```text
6>C:\Users\runner-user\.nuget\packages\gitinfo\3.5.0\build\GitInfo.targets(864,5): error MSB3491: Could not write lines to file "C:\runner-user\builds\plyD-vjk\0\mygroup\myproject1\/obj/GitInfo.cache". The process cannot access the file 'C:\runner-user\builds\plyD-vjk\0\mygroup\myproject1\obj\GitInfo.cache' because it is being used by another process. [C:\runner-user\builds\plyD-vjk\0\mygroup\myproject1\App1\App1.vbproj]
Build FAILED.
```

## Version Info

GitInfo 3.5.0

<!-- sl -->
[![Back this issue](https://sponsorlink.blob.core.windows.net/public/separator.png "Back this issue")](https://www.devlooped.com/SponsorLink/github/issues/?s=devlooped)
[![Back this issue](https://img.shields.io/badge/backed-%240-EA4AAA?logo=githubsponsors "Back this issue")](https://www.devlooped.com/SponsorLink/github/issues/?s=devlooped)
<!-- sl -->

--- a/changelog.md
+++ b/changelog.md
@@ -5,6 +5,7 @@
 :bug: Fixed bugs:
 
 - Assemblies of transitive nuget dependencies are placed into the build folder [\#389](https://github.com/devlooped/GitInfo/issues/389)
+- Build occasionally fails when using shared GitInfo.cache with parallel builds (msbuild /m) [\#390](https://github.com/devlooped/GitInfo/issues/390)
 
 ## [v3.6.0](https://github.com/devlooped/GitInfo/tree/v3.6.0) (2025-10-17)
 

--- a/src/GitInfo/build/GitInfo.targets
+++ b/src/GitInfo/build/GitInfo.targets
@@ -337,12 +337,12 @@
       <Output ItemName="_GitInput" TaskParameter="Include" />
     </CreateItem>
 
-    <Delete Files="$(_GitInfoFile)" Condition="Exists('$(_GitInfoFile)') and '$(SkipReadGitCache)' == 'true'" />
+    <Delete Files="$(_GitInfoFile)" Condition="Exists('$(_GitInfoFile)') and '$(SkipReadGitCache)' == 'true'" ContinueOnError="true" />
   </Target>
 
   <!-- If the inputs/outputs are outdated, clear the cache. -->
   <Target Name="_GitClearCache" Inputs="@(_GitInput)" Outputs="$(_GitInfoFile)" Condition="Exists('$(_GitInfoFile)')">
-    <Delete Files="$(_GitInfoFile)" />
+    <Delete Files="$(_GitInfoFile)" ContinueOnError="true" />
   </Target>
 
   <Target Name="_GitReadCache" Condition="Exists('$(_GitInfoFile)') and '$(SkipReadGitCache)' != 'true'">
@@ -859,9 +859,9 @@
       <_GitInfoFileDir>$([System.IO.Path]::GetDirectoryName('$(_GitInfoFile)'))</_GitInfoFileDir>
     </PropertyGroup>
 
-    <MakeDir Directories="$(_GitInfoFileDir)" Condition="!Exists('$(_GitInfoFileDir)')" />
+    <MakeDir Directories="$(_GitInfoFileDir)" Condition="!Exists('$(_GitInfoFileDir)')" ContinueOnError="true" />
 
-    <WriteLinesToFile File="$(_GitInfoFile)" Lines="$(_GitInfoContent)" Overwrite="true" />
+    <WriteLinesToFile File="$(_GitInfoFile)" Lines="$(_GitInfoContent)" Overwrite="true" ContinueOnError="true" />
 
     <ItemGroup>
       <FileWrites Include="$(_GitInfoFile)" />

